govc: Add feature to pass fullname and org for windows vm customization

rubinthomasdev opened the pull request:
## Description

Added two new flags to pass the `fullname` and `org` for windows vm customization mandatory parameters.

Closes: #3443

## Type of change

Please mark options that are relevant:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] This change requires a documentation update
- [ ] Build related change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide
instructions so we can reproduce. If applicable, please also list any relevant
details for your test configuration.

- [x] run below command to test that the vm is getting customized successfully
```
govc vm.customize -vm=<windows-vm-name> -u="<vcenter-url>" -dc="<datacenter-name>" -ip=<vm-ip> -netmask=<netmask> -gateway=<gateway-ip> -name=<windows-vm-name> -type=Windows -domain="<domain-name>" -username="firsname lastname" -org="myorg"
```

## Checklist:

- [x] My code follows the `CONTRIBUTION` [guidelines] of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

[guidelines]: https://github.com/vmware/govmomi/blob/main/CONTRIBUTING.md

dougm reviewed (commented):
Thanks @rubinthomasdev , please run `make doc` and include the update to govc/USAGE.md
Then we can merge once the CLA is approved

dougm commented on govc/vm/customize.go:
```diff
@@ -246,6 +250,18 @@ func (cmd *customize) Run(ctx context.Context, f *flag.FlagSet) error {
 		}
 	}
 
+	if cmd.username != "" {
```
This check and the one below isn't needed, since the UserData fields being set are the same type (`string`) and same default value (`""`) as the cmd flags. Looking at the existing code, there's another check than can be removed, you could then add the new cmd fields there:
```patch
diff --git a/govc/vm/customize.go b/govc/vm/customize.go
index 356ab6f2..82762486 100644
--- a/govc/vm/customize.go
+++ b/govc/vm/customize.go
@@ -211,12 +211,12 @@ func (cmd *customize) Run(ctx context.Context, f *flag.FlagSet) error {
 	sysprep, isWindows := spec.Identity.(*types.CustomizationSysprep)
 	linprep, _ := spec.Identity.(*types.CustomizationLinuxPrep)
 
-	if cmd.domain != "" {
-		if isWindows {
-			sysprep.Identification.JoinDomain = cmd.domain
-		} else {
-			linprep.Domain = cmd.domain
-		}
+	if isWindows {
+		sysprep.Identification.JoinDomain = cmd.domain
+		// add username/org fields below
+		sysprep.UserData.{...} = ...
+	} else {
+		linprep.Domain = cmd.domain
 	}
 
 	if len(cmd.dnsserver) != 0 {

```

rubinthomasdev commented:
Thank you @dougm for the review. I have done the following changes based on the review : 
1. run `make doc` to update the govc doc
2. removed the unnecessary check on empty strings and restructured the code based on the example above (thank you for the example).

dougm commented on govc/vm/customize.go:
```diff
@@ -75,6 +77,8 @@ func (cmd *customize) Register(ctx context.Context, f *flag.FlagSet) {
 	f.Var(&cmd.dnssuffix, "dns-suffix", "DNS suffix list")
 	cmd.dnssuffix = nil
 	f.StringVar(&cmd.kind, "type", "Linux", "Customization type if spec NAME is not specified (Linux|Windows)")
+	f.StringVar(&cmd.username, "username", "", "for windows vms, full name of the end user in firstname lastname format")
```
By convention, flag help text starts w/ upper-case 'For ...', how about prefix these both with:
> Windows only: full name of ...

rubinthomasdev replied:
Thank you for the review and suggestion. I have updated the description and the go doc accordingly.

dougm reviewed (commented):
One minor request on the doc. Could you also squash the commits and force push? We don't have the rebase option enabled for this repo, merge only. Thanks @rubinthomasdev

rubinthomasdev commented:
@dougm  Thank you for the suggestion. This makes the document more user friendly with `Windows only :` tag added. I have added the update and squashed and force pushed the changes.

dougm reviewed (approved):
Thanks @rubinthomasdev !
